mesonbuild/wrap/wrap.py

```diff
@@ -86,29 +95,45 @@ def has_patch(self):
         return 'patch_url' in self.values
 
 class Resolver:
-    def __init__(self, subdir_root):
+    def __init__(self, subdir_root, wrap_mode=WrapMode(1)):
+        self.wrap_mode = wrap_mode
         self.subdir_root = subdir_root
         self.cachedir = os.path.join(self.subdir_root, 'packagecache')
 
     def resolve(self, packagename):
-        fname = os.path.join(self.subdir_root, packagename + '.wrap')
-        dirname = os.path.join(self.subdir_root, packagename)
-        try:
-            if os.listdir(dirname):
-                # The directory is there and not empty? Great, use it.
+        # Check if the directory is already resolved
+        dirname = Path(os.path.join(self.subdir_root, packagename))
+        subprojdir = os.path.join(*dirname.parts[-2:])
+        if dirname.is_dir():
+            if (dirname / 'meson.build').is_file():
+                # The directory is there and has meson.build? Great, use it.
                 return packagename
-            else:
-                mlog.warning('Subproject directory %s is empty, possibly because of an unfinished'
-                             'checkout, removing to reclone' % dirname)
-                os.rmdir(dirname)
-        except NotADirectoryError:
-            raise RuntimeError('%s is not a directory, can not use as subproject.' % dirname)
-        except FileNotFoundError:
-            pass
+             # Is the dir not empty and also not a git submodule dir that is
```

Comment:
__[flake8]__
* [E114] indentation is not a multiple of four (comment)

<!-- Comment by [SideCI](https://sideci.com) -->